Firestore/core/src/firebase/firestore/util/async_queue.h

```diff
@@ -95,11 +96,30 @@ class AsyncQueue {
   // be called by a previously enqueued operation when it is run (as a special
   // case, destructors invoked when an enqueued operation has run and is being
   // destroyed may invoke `Enqueue`).
+  //
+  // After the shutdown process has initiated (`is_shutting_down()` is true),
+  // calling `Enqueue` will return, but the operation will *NOT* be run.
   void Enqueue(const Operation& operation);
 
+  // Like `Enqueue`, but also starts the shutdown process. Once the shutdown
+  // process has started, this queue will not run any operations requested
+  // via `Enqueue*` methods. They will not report error, they will simply
```

Comment:
Optional: similarly, this can be expressed shorter as `Once the shutdown process has started, calling any Enqueue* method becomes a no-op`.

Reply:
Done.